src/pip/_internal/commands/search.py

```diff
@@ -23,6 +20,17 @@
 from pip._internal.utils.compat import get_terminal_size
 from pip._internal.utils.logging import indent_log
 from pip._internal.utils.misc import write_output
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from optparse import Values
+    from typing import Any, List, Dict, Optional
+
+    # This type is actually "Dict[str, Union[str, List[str]]]"
+    # because the 'versions' key in the dictionary is a list.
+    # To get rid of Any type, the logic for transform_hits can be refactored to
+    # handle str and List[str] values separately
+    TransformedHit = Any
```

Comment:
This may work:

```suggestion
    from typing_extensions import TypedDict
    TransformedHit = TypedDict(
        'TransformedHit', 
        {'name': str, 'summary': str, 'versions': List[str]},
    )
```

Reply:
TypedDict is pretty cool tool to have when adding type annotations. Fixed in https://github.com/pypa/pip/pull/8265/commits/4bb3d7b399dd13d1081e7150db0d674639655bc8, although I had to do a hack with `summary`. variable in `print_results` to make the `TypedDict` work. Please do suggest is there a way to do that better.